subsys/testsuite/busy_sim/busy_sim.c

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <drivers/entropy.h>
+#include <drivers/counter.h>
+#include <drivers/gpio.h>
+#include "busy_sim.h"
+#include <hal/nrf_gpio.h>
+#include <sys/ring_buffer.h>
+
+struct busy_sim {
+	uint32_t idle_avg;
+	uint32_t active_avg;
+	uint16_t idle_delta;
+	uint16_t active_delta;
+	uint32_t us_tick;
+	const struct device *entropy;
+	const struct device *counter;
+	const struct device *gpio_port;
```

Comment:
These fields (and `pin` as well) can be initialized at compile time, so you could extract those to a separate `const` structure that could be placed in flash.

Reply:
This should use a gpio_dt_spec, also. Optional properties like active-gpios can be initialized using GPIO_DT_SPEC_GET_OR().

Reply:
> This should use a gpio_dt_spec, also. Optional properties like active-gpios can be initialized using GPIO_DT_SPEC_GET_OR().

@nordic-krch did you see this one? Just checking, since you've been pushing other updates.